secretmanager/api/SecretManagerSample/Program.cs

```diff
@@ -0,0 +1,381 @@
+            var request = new GetSecretVersionRequest
+            {
+                Name = name,
+            };
+
+            // Get the secret version.
+            var version = client.GetSecretVersion(request);
+            Console.WriteLine($"Secret version {version.Name}, state {version.State}");
+        }
+        // [END secretmanager_get_secret_version]
+
+        // [START secretmanager_get_secret]
+        public static void GetSecret(string name)
+        {
+            // string name = "projects/my-project/secrets/my-secret";
+
+            SecretManagerServiceClient client = SecretManagerServiceClient.Create();
+
+            // Create the request.
+            var request = new GetSecretRequest
+            {
+                Name = name,
+            };
+
+            // Get the secret.
+            var secret = client.GetSecret(request);
+            Console.WriteLine($"Secret {secret.Name}, replication {secret.Replication.ReplicationCase}");
+        }
+        // [END secretmanager_get_secret]
+
+        // [START secretmanager_list_secret_versions]
+        public static void ListSecretVersions(string parent)
+        {
+            // string parent = "projects/my-project/secrets/my-secret";
+
+            SecretManagerServiceClient client = SecretManagerServiceClient.Create();
+
+            // Create the request.
+            var request = new ListSecretVersionsRequest
+            {
+                Parent = parent,
+            };
+
+            // List all versions and their state.
+            foreach (var version in client.ListSecretVersions(request))
+            {
+              Console.WriteLine($"Secret version {version.Name}, {version.State}");
+            }
+        }
+        // [END secretmanager_list_secret_versions]
+
+        // [START secretmanager_list_secrets]
+        public static void ListSecrets(string parent)
+        {
+            // string parent = "projects/my-project";
+
+            SecretManagerServiceClient client = SecretManagerServiceClient.Create();
+
+            // Create the request.
+            var request = new ListSecretsRequest
+            {
+                Parent = parent,
+            };
+
+            // List all secrets in the project.
+            foreach (var secret in client.ListSecrets(request))
+            {
+              Console.WriteLine($"Secret {secret.Name}");
+            }
+        }
+        // [END secretmanager_list_secrets]
+
+        // [START secretmanager_update_secret]
+        public static void UpdateSecret(string name)
+        {
+            // string name = "projects/my-project/secrets/my-secret";
+
+            SecretManagerServiceClient client = SecretManagerServiceClient.Create();
+
+            // Create the request.
+            var request = new UpdateSecretRequest
+            {
+                Secret = new Secret
+                {
+                  Name = name,
+                  // Labels = new Dictionary<string, string>
```

Comment:
Help/guidance here would be appreciated. Labels is readonly, so I'm confused how to update them. Is that a bug in the generated library?

Reply:
No - the property is readonly, but it returns a mutable map. See https://googleapis.github.io/google-cloud-dotnet/docs/faq.html#how-can-i-modify-repeated-fields-and-maps-in-protobuf-messages